Library/Homebrew/formulary.rb

```diff
@@ -181,11 +181,6 @@ def self.load_formula_from_api(name, flags:)
         end
       end
 
-      if (keg_only_reason = json_formula["keg_only_reason"]).present?
-        reason = Formulary.convert_to_string_or_symbol keg_only_reason["reason"]
-        keg_only reason, keg_only_reason["explanation"]
-      end
```

Comment:
Can we do it this way:

```ruby
if (keg_only_reason = json_formula["keg_only_reason"]).present?
  reason = Formulary.convert_to_string_or_symbol keg_only_reason["reason"]
  reason.gsub!(HOMEBREW_PREFIX_PLACEHOLDER, HOMEBREW_PREFIX)
  keg_only reason, keg_only_reason["explanation"]
end
```

If possible, we should keep all things that might require API-specific handling to `Formulary::FormulaAPILoader` and out of `formula.json`. In general, we don't need to handle a user manually typing `$HOMEBREW_PREFIX` or interpolating `HOMEBREW_PREFIX_PLACEHOLDER`, so I think we shouldn't implement that substitution in `Formula::keg_only`

Reply:
I tried that first but it didn't substitute correctly, it was still showing `$HOMEBREW_PREFIX` as the placeholder? That's likely why we have `caveats` as a separate method in `Formulary` too?

Reply:
This modification seems to work for me:

```
if (keg_only_reason = json_formula["keg_only_reason"]).present?
  reason = Formulary.convert_to_string_or_symbol(keg_only_reason["reason"])
                    .gsub(HOMEBREW_PREFIX_PLACEHOLDER, HOMEBREW_PREFIX)
  keg_only reason, keg_only_reason["explanation"]
end
```

How are you testing it?